File: Harvest/Models/GardenerMessage.swift
import Foundation

struct GardenerMessage: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let role: String
    let content: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, role, content
        case userId = "user_id"
        case createdAt = "created_at"
    }
}

File: android/app/src/main/java/com/harvestglass/harvest/data/model/GardenerMessage.kt
package com.harvestglass.harvest.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors Harvest/Models/GardenerMessage.swift. */
@Serializable
data class GardenerMessage(
    val id: String,
    @SerialName("user_id") val userId: String,
    /** "user" or "assistant" — the DB column is `sender` with "gardener". */
    val role: String,
    val content: String,
    @SerialName("created_at") val createdAt: String? = null
)
